astroid/raw_building.py

```diff
@@ -157,10 +157,10 @@ def build_function(
         argsnode.defaults[-1].parent = argsnode
     if args:
         register_arguments(func)
-    return func
+    return func  # type: ignore[no-any-return]
 
 
-def build_from_import(fromname, names):
+def build_from_import(fromname: str | None, names: list[str]) -> nodes.ImportFrom:
```

Comment:
This is dead code and only used in tests. Shall we deprecate it? Perhaps in a follow up PR?

Either way this should be:
`fromname: str`. It makes no sense to have an `ImportFrom` without a from I think?

Reply:
I'd be happy to delete it!

`ImportFrom.fromname` is currently specified as `str | None`. That could be changed, but of course that would be out of scope.

Reply:
@Pierre-Sassoulas Do we consider this public?

I think it is better to add a deprecation and then remove in `3.0`. We are working towards that anyway.

Reply:
On the topic of `ImportFrom.fromnam`: we're probably going to change that in `3.0` which would mean we would need to revisit this function. Let's already fix this here and use `str` only.